pkg/processor/build/builder.go

```diff
@@ -1500,18 +1504,26 @@ func (b *Builder) resolveFunctionPathFromURL(functionPath string, codeEntryType
 			}
 		}
 
-		isArchive := codeEntryType == S3EntryType ||
-			codeEntryType == GithubEntryType ||
-			codeEntryType == ArchiveEntryType
-
-		// jar is an exception - we want it to remain compressed, as our java runtime processor expects to get it
-		isArchive = isArchive && !util.IsJar(functionPath)
-
 		tempDir, err := b.mkDirUnderTemp("download")
 		if err != nil {
 			return "", errors.Wrapf(err, "Failed to create temporary dir for download: %s", tempDir)
 		}
 
+		if codeEntryType == GitEntryType {
+			if err = b.cloneFunctionFromGit(tempDir, functionPath); err != nil {
```

Comment:
```suggestion
			if err := b.cloneFunctionFromGit(tempDir, functionPath); err != nil {
```

Reply:
not that is changes something critical, its just, if not need to lead `err`, avoid it